### ISSUE TITLE
Crash with Water Mobs

### ISSUE THREAD

**Issue by mxnmnm** [capnkirok/Inventory-Pets#298]
I tried to use the saddle pet on a sea serpent and this happened.
https://pastebin.com/Kis4msLK

**Comment by mxnmnm** [comment]
I can't log back into the world (good thing I use a throwaway world in case something like this happened lol) without crashing

**Comment by mxnmnm** [comment]
Another crash happened when I rode a siren (that I spawned on land) into the water https://pastebin.com/nPBdWnc5

**Comment by mxnmnm** [comment]
Ok so after more experimenting this seems to happen to a lot of water mobs such as fish from just a few fish (this happened on dry land) https://pastebin.com/91R2tne9

**Comment by mxnmnm** [comment]
accidentally closed it lol

**Comment by capnkirok** [comment]
Yeah, I made the assumption that other mods would all extend EntityWaterMob. Offending code:
```
if (creature instanceof EntityGuardian || creature instanceof EntitySquid || creature instanceof EntityWaterMob) {
//Do nothing as water creatures don't have this
} else {
creature.getNavigator().getNodeProcessor().setCanSwim(true);
}
```

I'll need to add a special null check for Custom Water Mobs.

**Comment by capnkirok** [comment]
Ok, this is fixed. Let's see if there any more issues.

By the way, you still won't be able to ride mobs in the water, bc Minecraft specifically forbids it and will dismount the player. I don't know why and couldn't figure a way around it.

But with the fix, at least it won't crash.

**Comment by mxnmnm** [comment]
I know you can ride a hippocampus from ice and fire.  Maybe you should check out the code and see how they did it and try to use the same method.  Also on a side-note some flying mobs freak out when you try to mount them with the flying saddle such as dragonflies from familiar fauna.

**Comment by capnkirok** [comment]
Alright this is fixed in 2.0.1.

Also, I tested the Hippocampus... it works because it rides above water, and it's also a custom mob.

Here's the code straight from Minecraft:

 ```
   /**
     * If the rider should be dismounted from the entity when the entity goes under water
     *
     * @param rider The entity that is riding
     * @return if the entity should be dismounted when under water
     */
    public boolean shouldDismountInWater(Entity rider)
    {
        return this instanceof EntityLivingBase;
    }
```

Basically, if it's any Entity Living in Minecraft, it automatically dismounts if you try to go under water. But, I could fix this in Animania, so thanks for bringing it up.
